assets/src/edit-story/components/library/mediaLibrary.js

```diff
@@ -208,163 +193,65 @@ function MediaLibrary({ onInsert }) {
-        uploadVideoFrame(videoId, src, videoEl.id);
-      }
-
-      return videoEl;
-    }
-    return null;
-  };
-
-  /**
-   * Get a formatted element for different media types.
-   *
-   * @param {Object} mediaEl Attachment object
-   * @param {number} width      Width that element is inserted into editor.
-   * @return {null|*}          Element or null if does not map to video/image.
-   */
-  const getMediaElement = (mediaEl, width) => {
-    const { src, oWidth, oHeight, mimeType } = mediaEl;
-    const origRatio = oWidth / oHeight;
-    const height = width / origRatio;
-    if (allowedImageMimeTypes.includes(mimeType)) {
-      return (
-        <Image
-          key={src}
-          src={src}
-          width={width}
-          height={height}
-          loading={'lazy'}
-          onClick={() => insertMediaElement(mediaEl, width)}
-        />
-      );
-    } else if (allowedVideoMimeTypes.includes(mimeType)) {
-      return (
-        <Video
-          key={src}
-          width={width}
-          height={height}
-          onClick={() => insertMediaElement(mediaEl, width)}
-          onMouseEnter={(evt) => {
-            evt.target.play();
-          }}
-          onMouseLeave={(evt) => {
-            evt.target.pause();
-            evt.target.currentTime = 0;
-          }}
-        >
-          <source src={src} type={mimeType} />
-        </Video>
-      );
-    }
-    return null;
-  };
-
   return (
-    <Dropzone>
-      <Header>
-        <Title>
-          {__('Media', 'web-stories')}
-          {(!isMediaLoaded || isMediaLoading) && <Spinner />}
-        </Title>
-        <UploadButton
-          onClose={onClose}
-          onSelect={onSelect}
-          buttonCSS={buttonStyles}
-        />
-      </Header>
-
-      <SearchField>
-        <Icon icon="search" />
-        <Search
-          value={searchTerm}
-          placeholder={__('Search Media', 'web-stories')}
-          onChange={onSearch}
-        />
-      </SearchField>
-
-      <FilterButtons>
-        {FILTERS.map(({ filter, name }, index) => (
-          <FilterButton
-            key={index}
-            active={filter === mediaType}
-            onClick={() => onFilter(filter)}
-          >
-            {name}
-          </FilterButton>
-        ))}
-      </FilterButtons>
-
-      {isMediaLoaded && !media.length ? (
-        <Message>{__('No media found', 'web-stories')}</Message>
-      ) : (
-        <Container>
-          <Column>
-            {media.map((mediaEl, index) => {
-              return isEven(index)
-                ? getMediaElement(mediaEl, DEFAULT_WIDTH)
-                : null;
-            })}
-          </Column>
-          <Column>
-            {media.map((mediaEl, index) => {
-              return !isEven(index)
-                ? getMediaElement(mediaEl, DEFAULT_WIDTH)
-                : null;
-            })}
-          </Column>
-        </Container>
-      )}
-    </Dropzone>
+    <DropZoneProvider>
```

Comment:
Perhaps instead of having two `DropZoneProvider`s, we could wrap the whole app to one provider. `DropZoneProvider` takes care of detecting which dropzone is being hovered over, there shouldn't be a need for having two. Unless it's added twice for some other reason?

Reply:
Not opposed to this, but if it's a lot of work I definitely recommend splitting that optimization into its own PR.